### PR TITLE
[front] fix: use client side pagination when needed

### PR DESCRIPTION
**Pull request by tdraier**
## Description

- Disable server-side pagination on managed datasource
- Remove default value for pagination on content-nodes
- Remove infinite scroll in trees as it's only for managed datasources
- Fixed pagination glitches when running client side
- Use client side pagination when using filter 

## Deploy Plan

front

### CODE REVIEW

File: front/components/DataSourceViewResourceSelectorTree.tsx [hunk 1]
```diff
@@ -15,9 +19,13 @@ import { useEffect, useState } from "react";
 
 import { RequestOrAddDataFromDataSourceModal } from "@app/components/data_source/RequestOrAddDataFromDataSourceModal";
 import DataSourceViewDocumentModal from "@app/components/DataSourceViewDocumentModal";
-import { InfiniteScroll } from "@app/components/InfiniteScroll";
 import { getVisualForContentNode } from "@app/lib/content_nodes";
+<<<<<<< HEAD
```
Comment (spolu) [inline front/components/DataSourceViewResourceSelectorTree.tsx:23]: :boom:

File: front/components/vaults/VaultDataSourceViewContentList.tsx [hunk 2]
```diff
@@ -205,6 +209,14 @@ export const VaultDataSourceViewContentList = ({
     [setPagination, setViewType, viewType, pagination.pageSize]
   );
 
+  const isServerPagination =
```
Comment (spolu) [inline front/components/vaults/VaultDataSourceViewContentList.tsx:212]: for managed data source we won't get the search then?

**Review by spolu — APPROVED** [review]
Looks very reasonable some rebase noise left

File: front/components/vaults/VaultDataSourceViewContentList.tsx (hunk 2 above)
Comment (tdraier) [reply front/components/vaults/VaultDataSourceViewContentList.tsx:212]: Yes,  managed data sources are not paginated serverside, all data loaded in the table, so search works

**Pull request merged**